js.go

```diff
@@ -2766,6 +2776,20 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 	return msgs, nil
 }
 
+// newFetchInbox returns subject used as reply subject when sending pull requests
+// as well as request ID. For non-wildcard subject, request ID is empty and
+// passed subject is not transformed
+func newFetchInbox(subj string) (string, string) {
+	if !strings.HasSuffix(subj, ".*") {
+		return subj, ""
+	}
+	reqID := nuid.Next()
+	var sb strings.Builder
```

Comment:
Maybe
```
return strings.Replace(subj, ".*", reqID, 1), reqID
```

Or 
```
reqID := nuid.Next()
trimmedSubj := strings.TrimSuffix(subj, ".*")
return trimmedSubj + reqID, reqID
```
?

Reply:
I was just being aware of performance here, as `Fetch()` might be invoked rather often (`strings.Builder` is more efficient than `+` or `fmt.Sprintf()`). Since we only concatenate 2 strings here, it probably does not make much difference, but I don't see clear benefit in changing it to `+` or `strings.Replace()` (which has to search the string).

Reply:
This PR is kind of rolling back the changes from this PR, which I think it is fine since it was intended as an optimization but I think it is better to have an inbox per fetch request so that they are isolated: https://github.com/nats-io/nats.go/commit/cfef5e6c50fe1ba7f6574f3ab9ea03c2ee58a3a9
This is how it was done before to generate these fetch reply inboxes in a fast way: https://github.com/nats-io/nats.go/blob/74d305fc3b1f5b6eafed5ab1a47d79e3540aa9d0/js.go#L867-L882

Reply:
Yeah, the original is faster, though I doubt it will have any noticeable impact. Let's leave it.